src/Hilke.KineticConvolution/DoubleAlgebraicNumber/DoubleConverter.cs

```diff
@@ -0,0 +1,249 @@
+
+        public TAlgebraicNumber FromDouble(double number) =>
+            AlgebraicNumberFactory.AlgebraicNumberCalculator.CreateConstant(number);
+
+        public double ToDouble(TAlgebraicNumber number) =>
+            AlgebraicNumberFactory.AlgebraicNumberCalculator.ToDouble(number);
+
+        public Point<TAlgebraicNumber> FromDouble(Point<double> point)
+        {
+            if (point is null)
+            {
+                throw new ArgumentNullException(nameof(point));
+            }
+
+            return AlgebraicNumberFactory.CreatePoint(FromDouble(point.X), FromDouble(point.Y));
+        }
+
+        public Point<double> ToDouble(Point<TAlgebraicNumber> point)
+        {
+            if (point is null)
+            {
+                throw new ArgumentNullException(nameof(point));
+            }
+
+            return DoubleFactory.CreatePoint(ToDouble(point.X), ToDouble(point.Y));
+        }
+
+        public Direction<TAlgebraicNumber> FromDouble(Direction<double> direction)
+        {
+            if (direction is null)
+            {
+                throw new ArgumentNullException(nameof(direction));
+            }
+
+            return AlgebraicNumberFactory.CreateDirection(FromDouble(direction.X), FromDouble(direction.Y));
+        }
+
+        public Direction<double> ToDouble(Direction<TAlgebraicNumber> direction)
+        {
+            if (direction is null)
+            {
+                throw new ArgumentNullException(nameof(direction));
+            }
+
+            return DoubleFactory.CreateDirection(ToDouble(direction.X), ToDouble(direction.Y));
+        }
+
+        public DirectionRange<TAlgebraicNumber> FromDouble(DirectionRange<double> range)
+        {
+            if (range is null)
+            {
+                throw new ArgumentNullException(nameof(range));
+            }
+
+            return AlgebraicNumberFactory.CreateDirectionRange(
+                FromDouble(range.Start),
+                FromDouble(range.End),
+                range.Orientation);
+        }
+
+        public DirectionRange<double> ToDouble(DirectionRange<TAlgebraicNumber> range)
+        {
+            if (range is null)
+            {
+                throw new ArgumentNullException(nameof(range));
+            }
+
+            return DoubleFactory.CreateDirectionRange(
+                ToDouble(range.Start),
+                ToDouble(range.End),
+                range.Orientation);
+        }
+
+        public Segment<TAlgebraicNumber> FromDouble(Segment<double> segment)
+        {
+            if (segment is null)
+            {
+                throw new ArgumentNullException(nameof(segment));
+            }
+
+            return AlgebraicNumberFactory.CreateSegment(
+                FromDouble(segment.Start),
+                FromDouble(segment.End),
+                segment.Weight);
+        }
+
+        public Segment<double>? ToDouble(Segment<TAlgebraicNumber> segment)
```

Comment:
```cs

        public Segment<double> ToDouble(Segment<TAlgebraicNumber> segment)
        {
            if (segment is null)
            {
                throw new ArgumentNullException(nameof(segment));
            }

            return DoubleFactory.CreateSegment(
                           ToDouble(segment.Start),
                           ToDouble(segment.End),
                           segment.Weight);
        }
````

Reply:
If we are unlucky, it can happen that converting `start` and `end` to `double` brings  `start == end`, which does not lead to a viable `Segment<double>`. That's why in this case I return `null`

Reply:
I changed the method's name to `TryToDouble` and modified the handling of converting to `DirectionRange` to double  as a similar issue can be encountered